properties.h

```diff
@@ -1,6 +1,22 @@
 #ifndef __properties_H_
 #define __properties_H_
 
+#ifdef LIBPROPERTIES_EXPORT
+#ifdef _WIN32
+#define LIBPROPERTIES_EXTERN __declspec(dllexport)
+#else
+#define LIBPROPERTIES_EXTERN __attribute__((visibility("default")))
+#endif
+#elif !defined(LIBPROPERTIES_STATIC)
+#ifdef _WIN32
+#define LIBPROPERTIES_EXTERN __declspec(dllimport)
+#else
+#define LIBPROPERTIES_EXTERN
+#endif
+#else
+#define LIBPROPERTIES_EXTERN
```

Comment:
We'd better use the name `LIBPROPERTIES_API`.

Reply:
I will make the change at the next opportunity

Reply:
I have solved.